### PR TITLE
Adding circle.md guide

### PR DESCRIPTION
**Pull request by csidyel**
## What
A new CircleCI migration page using the GitHub Actions page as a baseline.

## Checklist: 
 

Use a checklist to confirm:
- [x] The branch is up-to-date with the main branch.
- [x] Update follows the docs [style guide](../../docs/docs-contributing/STYLE_GUIDE.md).
- [x] Changes have been tested locally.

### CODE REVIEW

File: docs/docs/getting-started/migration/circle.md [hunk 1]
```diff
@@ -0,0 +1,550 @@
...
+      - my_job:
+          # highlight-start
+          context:
+            - awskey
+          # highlight-end
+```
+
+</TabItem>
+<TabItem value="semaphore" label="Semaphore YAML">
+
```
Comment (TomFern) [inline docs/docs/getting-started/migration/circle.md:310]:
```suggestion

On Semaphore, we create the [secret](../../using-semaphore/secrets) at the organization or project level and activate it on a block. 

The secret's contents are automatically injected as environment variables in all jobs in that block.
```

Comment (TomFern) [reply docs/docs/getting-started/migration/circle.md:310]: I think we should mention that secrets must already be created. We can use the same text as in the GUI tab.

**Review by TomFern — CHANGES_REQUESTED** [review]
@csidyel thank you. I think this looks great. There is just one comment to review.

File: docs/docs/getting-started/migration/circle.md (hunk 1 above)
Comment (csidyel) [reply docs/docs/getting-started/migration/circle.md:310]: I think it's solved now. wdyt?

Comment (TomFern) [reply docs/docs/getting-started/migration/circle.md:310]: Yeah, totally!

**Review by TomFern — APPROVED** [review]

**Pull request merged**